pkg/istio/authorizationpolicy/controller.go

```diff
@@ -117,6 +118,15 @@ func (c *Controller) Run(stopCh <-chan struct{}) {
 	go c.authzpolicyIndexInformer.Run(stopCh)
 	go c.resync(stopCh)
 
+	if c.configStoreCache.HasSynced() {
+		// If the service is switching back from Authorization Policy Enabled back to SR/SRB delete the
+		// existing Authorization Policy associated to the service
+		err := c.cleanUpStaleAP()
+		if err != nil {
+			log.Errorf("Error while running cleanUpStaleAP: %v", err.Error())
```

Comment:
If clean up failed with problem, do we want to panic exiting the controller?

Reply:
Currently, it does not but I can change it to error out in case the cleanup failed. Can not think of any cons with that. Thoughts @mdhume @mcieplak ?

Reply:
I think it makes sense to panic, this will at least alert us that something is wrong. Otherwise if we move on then we might run into a weird state with some authz policies still on the clusters that have now moved back to sr / srb.